vignettes/incidence2.Rmd

```diff
@@ -0,0 +1,137 @@
+---
+title: "incidence2"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{incidence2}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  fig.align = "center",
+  fig.width = 7,
+  fig.height = 5
+)
+```
+
+## What does it do?
+
+**incidence2** is an R package that implements functions and classes to compute,
+handle and visualise *incidence* data.
+
+### Overview
+
+The main feature of the package is the `incidence()` function which works on
+both linelist and pre-aggregated data sets. It returns an object of class
+`<incidence>`. This is a data frame subclass with some additional invariants.
+That is, an `<incidence>` object must:
+  
+- have one column representing the date index (this does not need to be a `date`
+  object but must have an inherent ordering over time);
+  
+- have one column representing the count variable (i.e. what is being counted)
+  and one variable representing the associated count;
+  
+- have zero or more columns representing groups;
+
+- not have duplicated rows with regards to the date and group variables.
+    
+### Additional functionality
+
+- `plot.incidence()`: generate simple plots with reasonable defaults.
+
+- `regroup()`: regroup incidence from different groups into one global incidence
+  time series.
+  
+- `keep_first()` and `keep_last()`: keep the rows corresponding to the first (or
+  last) set of grouped dates (ordered by time) from an `<incidence>` object.
+  
+- `complete_counts()`: ensure every possible combination of date and groupings
+  is given an explicit count.
```

Comment:
```suggestion
  is given an explicit count, adding zeros as needed.
```

Reply:
Makes sense. Will leave this open for moment and may do grammatical changes all in one commit.